src/System.Configuration.ConfigurationManager/src/System/Configuration/GenericEnumConverter.cs

```diff
@@ -32,35 +33,54 @@ public override object ConvertFrom(ITypeDescriptorContext ctx, CultureInfo ci, o
             try
             {
                 string value = (string)data;
-                if (string.IsNullOrEmpty(value)) throw new Exception();
+                bool valueIsInvalid = string.IsNullOrEmpty(value);
```

Comment:
Why not just `throw CreateExceptionForInvalidValue()` instead of setting a flag and having to test it again?